src/Microsoft.Restier.Core/Conventions/ConventionBasedEntitySetProcessor.cs

```diff
@@ -53,48 +56,130 @@ public Expression Process(QueryExpressionContext context)
             }
 
             var dataSourceStubReference = context.ModelReference as DataSourceStubModelReference;
-            if (dataSourceStubReference == null)
+            if (dataSourceStubReference != null)
+            {
+                var entitySet = dataSourceStubReference.Element as IEdmEntitySet;
+                if (entitySet == null)
+                {
+                    return null;
+                }
+
+                var collectionType = entitySet.Type as EdmCollectionType;
```

Comment:
better convert to `"I"EdmCollectionType` and `"I"EdmEntityType` because the model may not necessarily be `EdmModel`, e.g., `CsdlSemanticModel`.